### ISSUE TITLE
2.4.11 and 2.5.8 - Recommended viewport size for testing?

### ISSUE THREAD

**Issue by LiLoDavis** [w3c/wcag#1829]
Conformance with **2.4.11 Focus Appearance (Minimum)** and **2.5.8 Target Size (Minimum)** can depend on an element's height and width. Given that elements' dimensions can vary with changes in the viewport size and/or zoom setting, can you recommend a viewport size and zoom setting for testing these SC?

**Comment by jake-abma** [comment]
Hey @LiLoDavis not viewport sizes need to be tested, but all variations of a **"full page"** (they may be triggered by viewport changes though...)

See also: https://www.w3.org/TR/WCAG21/#cc2

> NOTE
> A full page includes each variation of the page that is automatically presented by the page for various screen sizes (e.g. variations in a responsive Web page). Each of these variations needs to conform (or needs to have a conforming alternate version) in order for the entire page to conform.

**Comment by alastc** [comment]
> can you recommend a viewport size and zoom setting for testing these SC?

For focus appearance, if you have sticky / layered content, I would test at each page variation (e.g. media query kicking in). It would apply to the 'obscured' bullet, and if you had shapes (rather than outlines or backgrounds) it might apply to the sizing for things which change shape at different viewport sizes.

For target size, as it's measured in CSS pixels it would only really be if the size or spacing of links deliberately changed that you'd need to test. 

So overall, there isn't really a recommended size to test at, as Jake mentioned you need to test different page variations. What I'd recommend is examining when the elements on the page are changing, and therefore might need testing.

**Comment by LiLoDavis** [comment]
Thank you, @jake-abma and @alastc. 
I was evaluating a page for target size when I realized that some text fields changed width automatically (to fit the space available) as I adjusted the width of the page. If I understand the SC correctly, as target sizes decrease below 24px, the target offset needs to increase in order to conform.

**Comment by alastc** [comment]
If a text-input gets smaller than 24px, you wouldn't fit more than a letter or two in it?!

**Comment by alastc** [comment]
Proposed response:

<hr>

Conformance should be tested across page variations, and the new focus-appearance and target-size SCs do not create specific scenarios where a particular viewport size is required.

If the pages are being tested down to 320px wide (for reflow), that would be a useful minimum size for testing the smallest page-variant.

**Comment by alastc** [comment]
The response above was approved: https://www.w3.org/2022/02/01-ag-minutes#item07
